src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -400,6 +402,28 @@ public virtual async Task<ConnectionCompleteParams> Connect(ConnectParams connec
             return completeParams;
         }
 
+        private async Task<ConnectionCompleteParams> TryOpenConnectionWithRetry(ConnectionInfo connectionInfo, ConnectParams connectionParams)
+        {
+            Boolean isRetriable = true;
+            int counter = 0;
+            ConnectionCompleteParams response = null;
+            while (isRetriable && counter <= MaxTries)

```

Comment:
@cheenamalhotra Is there a reason you know of why we can't/shouldn't use the built-in retry logic of SqlClient? @smartguest was this something you investigated before going with a custom retry solution?

https://docs.microsoft.com/en-us/sql/connect/ado-net/configurable-retry-logic?view=sql-server-ver16

Reply:
We need to evaluate all aspects of enabling it before we enable it, as the doc says, there are some limitations to it specially with transaction management, and it may cause disruptions rather than being useful. Retry logic is mainly for enterprise apps, where queries are in control.

Reply:
I did not investigate using SqlClient yet, but the OpenAsync DbConnection function that TryOpenConnection uses doesn't retry for it currently, so with working with the existing code before implementing SqlClient, this is the option I have taken.

Reply:
Alright, seems reasonable to just do this given how straightforward it is to implement the retry ourselves. I'd suggest adding a comment explaining this though for future context.